client/src/project/datasets/change/DatasetChange.container.js

```diff
@@ -220,7 +230,17 @@ function ChangeDataset(props) {
       .then(response => {
         if (response.data.error !== undefined) {
           handlers.setSubmitLoader({ value: false, text: "" });
-          handlers.setServerErrors(response.data.error.reason);
+          if (response.data.error.errorOnFileAdd) {
+            props.edit ?
+              handlers.setServerErrors(
+                <div>
+                  The dataset was edited but there was an error adding uploaded files to it, please try again.
```

Comment:
Is this an accurate explanation of the error situation?

```suggestion
                  Any metadata changes were successfully applied, but some uploaded files could not be added to the dataset. Please try adding files again.
```